pyat/at/collective/wake_object.py

```diff
@@ -165,3 +165,49 @@ def reswall(self, wcomp, length, rvac, conduct, beta, yokoya_factor=1):
                                       length, rvac, conduct, beta)
         else:
             raise AtError('Invalid WakeComponent: {}'.format(wcomp))
+
+
+def res_object(srange, wakecomp, frequency, qfactor, rshunt, beta,
+              yokoya_factor=1):
+    """
+    Method to build a resonator wake object
+    """
+    wake = Wake(srange)
+    wakecomp = numpy.atleast_1d(wakecomp)
+    try:
+        frequency = numpy.broadcast_to(frequency, (len(wakecomp), ))
+        qfactor = numpy.broadcast_to(qfactor, (len(wakecomp), ))
+        rshunt = numpy.broadcast_to(rshunt, (len(wakecomp), ))
+        yokoya_factor = numpy.broadcast_to(yokoya_factor, (len(wakecomp), )) 
+    except ValueError:
+        raise AtError('Wake object inputs should be either scalars '
+                      'or with shape (len(wakecomp), )')
+    for wc, fr, qf, rs, yk in zip(wakecomp, frequency, qfactor, rshunt, yokoya_factor):
+        wake.add(WakeType.RESONATOR, wc, fr, qf, rs, beta, yokoya_factor=yk)
+    return wake
+
+
+def longres_object(srange, frequency, qfactor, rshunt, beta):
+    """
+    Method to build a longitudinal resonator wake object
+    """
```

Comment:
I would prefer to implement this as a static method of the `Wake` class rather than an independent function:
```python
@staticmethod
def long_resonator(srange, frequency, qfactor, rshunt, beta):
```
A longitudinal resonator object would be created with:
```python
long_res = Wake.long_resonator(srange, frequency, qfactor, rshunt, beta)
```
This cleans the global namespace and allows better names.
And the same for the `res_object` and `reswall_object`.

Reply:
Again I am a bit confused, these methods return a `Wake` instance, does it make sense for a class to initialize itself in a static method? 

Other topic: why is python3.7 test in windows systematically failing?

Reply:
> Again I am a bit confused, these methods return a Wake instance, does it make sense for a class to initialize itself in a static method?

Of course, it makes sense. it's much cleaner ! That's factory methods.

Concerning the tests, the failure is due to a problem on GitHub actions for windows runners. They changed the way python 3.7 is built on windows, so that the file suffix for compiled extension is now `.pyd` instead of `.cp37-win_amd64.pyd`. It's probably a bug on their side, but the way PyAT gets the suffix has to be changed. The modification concerns 2 files:  `pyat/at.c` and `pyat/at/load/utils.py`. I have a working solution in the `turn_number` branch, but since it looks like a GitHub bug, I would rather disable the test on windows/python 3.7. For this, modify the `at/.github/python-tests.yml` file: replace

```
    strategy:
      matrix:
        python-version: ['3.6', '3.7', '3.8', '3.9', '3.10']
        os: [macos-latest, ubuntu-latest, windows-latest]
```
with
```
    strategy:
      matrix:
        python-version: ['3.6', '3.7', '3.8', '3.9', '3.10']
        os: [macos-latest, ubuntu-latest, windows-latest]
        exclude:
          - os: windows-latest
            python-version: '3.7'
```

Reply:
Sorry but I do not see it... maybe you mean a `@classmethod`? this I could understand but a static method no... you have to show me an example of what you have in mind

Reply:
NO, not a class method ! You need a static method ! Just move your factory functions inside the class, without  any change, and prefix them with "@staticmethod". They still behave as a normal functions, but  are encapsulated in the class for cleanliness. For instance, look at the` Lattice.load` static method (last lines of `load/allfiles.py`):

`ring = Lattice.load(filepath)` calls such a factory method (a static method of the Lattice class) to create a lattice object from a file name.

Here you would call
```python
long_res = Wake.long_resonator(srange, frequency, qfactor, rshunt, beta)
```
to create a wake object. With this, you do not introduce a new function in the global namespace, and the "Wake" prefix clearly shows the kind of object you create.

A class method would have a "class" 1<sup>st</sup> argument which is useless for this purpose.

Reply:
Well a class method would result in exactly the same syntax and cleanliness but also handle inheritance no? So it seems cleaner to me while I do not see any drawback.

Anyway it is just a question of taste, I can implement what you suggest